components/brave_wallet_ui/common/hardware/ledgerjs/sol_ledger_bridge_keyring.ts

```diff
@@ -0,0 +1,116 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { LEDGER_HARDWARE_VENDOR } from 'gen/brave/components/brave_wallet/common/brave_wallet.mojom.m.js'
+import TransportWebHID from '@ledgerhq/hw-transport-webhid'
+import Transport from '@ledgerhq/hw-transport'
+import Sol from '@ledgerhq/hw-app-solana'
+import { BraveWallet } from '../../../constants/types'
+import { LedgerSolanaKeyring } from '../interfaces'
+import { HardwareVendor } from '../../api/hardware_keyrings'
+import {
+  GetAccountsHardwareOperationResult,
+  HardwareOperationResult,
+  SignHardwareMessageOperationResult,
+  SignHardwareTransactionOperationResult
+} from '../types'
+
+import { hardwareDeviceIdFromAddress } from '../hardwareDeviceIdFromAddress'
+
+export default class SolanaLedgerKeyring implements LedgerSolanaKeyring {
+  private app?: Sol
+  private transport?: Transport
+  private deviceId: string
+
+  coin = (): BraveWallet.CoinType => {
+    return BraveWallet.CoinType.SOL
+  }
+
+  type = (): HardwareVendor => {
+    return LEDGER_HARDWARE_VENDOR
+  }
+
+  getAccounts = async (from: number, to: number): Promise<GetAccountsHardwareOperationResult> => {
+    const unlocked = await this.unlock()
+    if (!unlocked.success || !this.app) {
+      return unlocked
+    }
+    from = (from < 0) ? 0 : from
+    const sol: Sol = this.app
+    const accounts = []
+    for (let i = from; i <= to; i++) {
+      const path = this.getPathForIndex(i)
+      const address = await sol.getAddress(path)
+      accounts.push({
+        address: '',
+        addressBytes: address.address,
+        derivationPath: path,
+        name: this.type(),
+        hardwareVendor: this.type(),
+        deviceId: this.deviceId,
+        coin: this.coin()
+      })
+    }
+    return { success: true, payload: [...accounts] }
+  }
+
+  isUnlocked = (): boolean => {
+    return this.app !== undefined && this.deviceId !== undefined
+  }
+
+  makeApp = async () => {
```

Comment:
Some of these functions could likely be refactored with the Filecoin/Ethereum ledger bridges. I didn't double check, but I'm pretty sure a lot of this logic is nearly similar based on what I remember when reviewing the Filecoin work. I wouldn't consider this a blocking issue, but something that I would think is worth tracking in an issue to refactor later.

Reply:
++

Reply:
Created an issue for it but I have limited capabilities in that repo, so if there's anything else that needs to be added to it for labels/assignees can you modify it? https://github.com/brave/brave-browser/issues/23554